ops/model.py

```diff
@@ -1361,6 +1362,17 @@ def _validate_action_result_key(key: str) -> None:
     Raises:
         ValueError: if a dict is passes with a key that fails to meet the format requirements.
     """
+    # Existing charms may well be passing dicts to set_result that contain dicts pre-formatted
+    # to use dotted notation having worked around this in the past. We will deprecate this in the
+    # future to avoid possible duplication of keys.
+    # TODO: Drop this in a future release
+    if '.' in key:
+        msg = ("a '.' was detected in the dict key: '{}' while formatting a dict passed to "
+               "ActionEvent.set_result(). Keys using the dotted notation will be deprecated in a "
```

Comment:
I believe they are *deprecated* now, but will be *rejected* in a later release.

Reply:
Fixed :) Thanks.